src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,337 @@
+            - Data from successive exposures is added as a new dimension, then returns a 3D array.
+            - Averaging across frames can be done by summing over this additional dimension.
+        """
+        data = np.array([])
+        for i in range(image_dataset.Frames):
+            frame = image_dataset.GetFrame(0, i)
+            new_frame = np.reshape(np.fromiter(frame.GetData(), dtype='uint16'), [frame.Width, frame.Height], order='F')
+            data = np.dstack((data, new_frame)) if data.size else new_frame
+        return data
+
+    #NOTE: May not need to call the 'start_acquisition_and_wait' method if you fix the 'FileNameGeneration' issue
+    def acquire(self) -> np.ndarray:
+        """
+        Acquires image data from the spectrometer.
+        """
+        self.file_setup()
+        self.start_acquisition_and_wait()
+        return self.process_acquired_data()
+
+    def close(self) -> None:
+        """
+        Closes the Lightfield application without saving the settings.
+        """
+        self._automation.Dispose()
+        logger.info('Closed AddInProcess.exe')
+
+class SpectrometerConfig():
+
+    MIN_WAVELENGTH_DIFFERENCE = 117
+
+    def __init__(self, experiment_name=None) -> None:
+        self._experiment_name = experiment_name
+        self.light = LightfieldApplicationManager(True)
+
+    def finalize(self) -> None:
+        """
+        Closes the Lightfield application without saving the settings.
+        """
+        self.light.close()
+        
+    def stop_scan(self) -> None:
+        """ 
+        Stop/Pause the current acquisition. If you click "Start" you will be able to continue the scan.
+        """
+        self.light.experiment.Stop()
+    
+    def get_wavelengths(self) -> np.ndarray:
+        """
+        Returns the wavelength calibration for a single frame.
+
+        According to the person who made the Lightfield sofware there is not current way to use the exact Lightfield settings
+        for step and glue. Will have to interpolate for now.
+        """
+        wavelength_array = np.fromiter(self.light.experiment.SystemColumnCalibration, np.float64)
+        return wavelength_array
+    
+    @property
+    def center_wavelength(self) -> float:
+        """
+        Returns the center wavelength in nanometers.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingCenterWavelength)
+
+    @center_wavelength.setter
+    def center_wavelength(self, nanometers: float) -> None:
+        """
+        Sets the spectrometer center wavelength to nanometers.
+        """
+        # The line below addresses bug where if step and glue is enabled it wont allow you to set the center wavelength.
+        self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEnabled, False)
+        self.light.set(lf.AddIns.SpectrometerSettings.GratingCenterWavelength, Double(nanometers))
+
+    @property
+    def experiment_name(self) -> Union[str, None]:
```

Comment:
explain what the experiment name is in a doc string, also, you can probably indicate that it will be either string or None by writing `str | None`

Reply:
When qt3-utils was developed, I started with python 3.9... for a reason I can't quite remember... maybe to support nidaq or known testing with nidaq. Anyways, 3.9 is the default python version on the QLM. The pipe `|` support for type hints wasn't introduced until 3.10. So we should stick with `Union`.

Reply:
Right. I forgot about that. Sounds good!